cpp/include/cuml/experimental/fil/detail/evaluate_tree.hpp

```diff
@@ -71,7 +73,11 @@ HOST DEVICE auto evaluate_tree(
     node += cur_node.child_offset(condition);
     cur_node = *node;
   } while (!cur_node.is_leaf());
-  return cur_node.template output<has_vector_leaves>();
+  if constexpr (evaluate_leaf) {
+    return static_cast<node_t const*>(node);
```

Comment:
Why is the `static_cast` needed here?

Reply:
Actually you're right, it's probably not necessary.